public/app/plugins/datasource/grafana-azure-monitor-datasource/types/index.ts

```diff
@@ -109,50 +111,55 @@ export interface AzureMetricDimension {
 }
 
 export interface AzureMetricQuery {
-  resourceGroup: string | undefined;
-  resourceName: string | undefined;
-  metricDefinition: string | undefined;
-  metricNamespace: string | undefined;
-  metricName: string | undefined;
+  resourceGroup?: string | undefined;
```

Comment:
if you have a `?` do you need this `| undefined` as well?

Reply:
Good catch - nope!!!